src/group_theory/perm/list.lean

```diff
@@ -342,6 +347,109 @@ begin
       rw [hl, nat.mod_eq_of_lt hk', h1, ←nat.add_mod, nat.succ_add] } }
 end
 
+lemma form_perm_apply_mem_eq_self_iff (hl : nodup l) (x : α) (hx : x ∈ l) :
+  form_perm l x = x ↔ length l ≤ 1 :=
+begin
+  obtain ⟨k, hk, rfl⟩ := nth_le_of_mem hx,
+  rw [form_perm_apply_nth_le _ hl, hl.nth_le_inj_iff],
+  cases hn : l.length,
+  { exact absurd k.zero_le (hk.trans_le hn.le).not_le },
+  { rw hn at hk,
+    cases (nat.le_of_lt_succ hk).eq_or_lt with hk' hk',
+    { simp [←hk', nat.succ_le_succ_iff, eq_comm] },
+    { simpa [nat.mod_eq_of_lt (nat.succ_lt_succ hk'), nat.succ_lt_succ_iff]
+        using k.zero_le.trans_lt hk' } }
+end
+
+lemma form_perm_apply_mem_ne_self_iff (hl : nodup l) (x : α) (hx : x ∈ l) :
+  form_perm l x ≠ x ↔ 2 ≤ l.length :=
+begin
+  rw [ne.def, form_perm_apply_mem_eq_self_iff _ hl x hx, not_le],
+  exact ⟨nat.succ_le_of_lt, nat.lt_of_succ_le⟩
+end
+
+lemma form_perm_apply_not_mem (l : list α) (x : α) (h : x ∉ l) :
+  form_perm l x = x :=
+begin
+  contrapose! h,
+  suffices : x ∈ {y | form_perm l y ≠ y},
+  { rw ←mem_to_finset,
+    exact support_form_perm_le' _ this },
+  simpa using h
+end
+
+lemma form_perm_ne_self_imp_mem (l : list α) (x : α) (h : form_perm l x ≠ x) :
+  x ∈ l :=
+begin
+  contrapose! h,
+  exact form_perm_apply_not_mem _ _ h
```

Comment:
Since the first like of `form_perm_apply_not_mem ` is `contrapose!`, maybe it makes sense to prove these in the other order.